pyintegration/deephaven2/table.py

```diff
@@ -55,6 +222,801 @@ def update(self, formulas: List[str]):
+        order based on the order_by column(s).
+
+        Args:
+            order_by (List[str], optional): the list of column names
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            return self._j_table.sortDescending(*order_by)
+        except Exception as e:
+            raise DHError(e, "table sort_descending operation failed.") from e
+
+    def reverse(self) -> Table:
+        """ The reverse method creates a new table with all of the rows from this table in reverse order.
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            return Table(j_table=self._j_table.reverse())
+        except Exception as e:
+            raise DHError(e, "table reverse operation failed.") from e
+
+    def sort(self, order_by: List[str], order: List[SortDirection] = []) -> Table:
+        """ The sort method creates a new table where (1) rows are sorted in a smallest to largest order based on the
+        order_by column(s) (2) where rows are sorted in the order defined by the order argument.
+
+.       Args:
+            order_by (List[str]): the names of the columns to be sorted on
+            order (List[SortDirection], optional): the corresponding sort directions for each sort column, default
+                is empty. In the absence of explicit sort directions, data will be sorted in the ascending order.
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+
+        def sort_pair(col, dir_):
+            return SortPair.descending(col) if dir_ == SortDirection.DESCENDING else SortPair.ascending(col)
+
+        try:
+            if order:
+                sort_pairs = [sort_pair(col, dir_) for col, dir_ in zip(order_by, order)]
+                return Table(j_table=self._j_table.sort(*sort_pairs))
+            else:
+                return Table(j_table=self._j_table.sort(*order_by))
+        except Exception as e:
+            raise DHError(e, "table sort operation failed.") from e
+
+    #
+    # Table operation category: Filter
+    #
+    def where(self, filters: List[str] = []) -> Table:
+        """ The where method creates a new table with only the rows meeting the filter criteria in the column(s) of
+        the table.
+
+        Args:
+            filters (List[str]): a list of filter condition expressions
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
+        try:
+            if not filters:
+                return Table(j_table=self._j_table.where())
+            else:
+                return Table(j_table=self._j_table.where(*filters))
+        except Exception as e:
+            raise DHError(e, "table where operation failed.") from e
+
+    def where_in(self, filter_table, cols: List[str]) -> Table:
```

Comment:
We can provide typing for filter_table, yeah?

Reply:
yes, done